pkg/config/config.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	ioutil "io/ioutil"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+)
+
+// ControllerConfiguration defines the functions necessary to parse a config file
+// and to configure the Options struct for the ctrl.Manager
+type ControllerConfiguration interface {
+	runtime.Object
+
+	// GetControllerConfiguration returns the versioned configuration
+	GetControllerConfiguration() (v1alpha1.ControllerConfigurationSpec, error)
+}
+
+// DeferredFileLoader is used to setup the file loader for manager.Options
+// and a ComponentConfig type
+type DeferredFileLoader interface {
```

Comment:
seems like this doesn't actually need to be an interface -- can just have a concrete struct with these methods (or, for that matter, just a struct with these fields)

Reply:
We're discussing this over Slack.

Reply:
Updated this to be a struct in the latest push.